### PR TITLE
Refine completions (doc fix, profile completions, more fish completions)

### PR DESCRIPTION
**Pull request by yu-ichiro**
Added some completion related changes

* Fixed README: `ghr shell bash | source` → `source <(ghr shell bash)`
* Added `--short` option to `ghr profile list` for completions
* Added profile completions to bash + Made open complete repos
* Removed file completions from fish + Added subcommand completions

### CODE REVIEW

File: resources/shell/bash/ghr-completion.bash [hunk 1]
```diff
@@ -48,13 +61,27 @@ __ghr_complete() {
...
   open)
-    COMPREPLY=($(__ghr_complete__static "${cword}" --help))
+    if [ "$COMP_CWORD" = 2 ]; then
+      COMPREPLY=($(__ghr_complete__repos "${cword}" --help))
+    elif [ "$COMP_CWORD" = 3 ]; then
+      COMPREPLY=($(compgen -c -- "${cword}"))
+    fi
     ;;
   path)
     COMPREPLY=($(__ghr_complete__repos "${cword}"))
```
Comment (siketyan) [inline resources/shell/bash/ghr-completion.bash:72]: `ghr open <tab><tab>` worked, but `ghr open siketyan<tab><tab>` did not work for me. And this is caused by `__ghr_complete__repos`, it is broken :cry:

**Review by siketyan — APPROVED** [review]

**Comment by siketyan** [comment]
Thank you for the awesome refinement! I am not good at completions at all, it helped me a lot ❤️

File: resources/shell/bash/ghr-completion.bash (hunk 1 above)
Comment (yu-ichiro) [reply resources/shell/bash/ghr-completion.bash:72]: thats wierd 🤔 gonna look in to it

Comment (yu-ichiro) [reply resources/shell/bash/ghr-completion.bash:72]:
```
$ choices=(abc bca cab)
$ compgen -W "${choices[*]}" -- "a"
abc
```

unfortunately compgen doesn't support partial matches 😢

**Pull request merged**